src/scripts/dungeons/DungeonRunner.ts

```diff
@@ -221,6 +221,17 @@ class DungeonRunner {
         return RouteHelper.listCompleted(possiblePokemon, includeShiny);
     }
 
+    public static isAchievementsComplete(dungeon: Dungeon) {
+        let result = true;
+        AchievementHandler.achievementList.forEach(achievement => {
+            if (achievement.property instanceof ClearDungeonRequirement && achievement.property.dungeonIndex === GameConstants.getDungeonIndex(dungeon.name) && !achievement.isCompleted()) {
+                result = false;
+                return true;
+            }
+        });
+        return result;
```

Comment:
Minor optimisations:
_Calculate the dungeon index outside of the loop, so we don't need to keep calling the same function._
_Use `.every` instead of `.forEach` so we stop as soon as we hit a false_
```suggestion
        // Calculate dungeon index
        const dungeonIndex = GameConstants.getDungeonIndex(dungeon.name);
        // Stop if we get a false
        AchievementHandler.achievementList.every(achievement => {
          return !(achievement.property instanceof ClearDungeonRequirement && achievement.property.dungeonIndex === dungeonIndex && !achievement.isCompleted());
        });
```
Could you apply a similar approach to the other achievement checks please.

Reply:
improvements added